src/main/kotlin/no/nav/familie/ef/sak/behandling/domain/Behandling.kt

```diff
@@ -37,6 +37,8 @@ data class Behandling(@Id
                       @Embedded(onEmpty = Embedded.OnEmpty.USE_EMPTY)
                       val sporbar: Sporbar = Sporbar(),
                       val resultat: BehandlingResultat,
+                      @Column("grunnbelopsdato")
+                      val grunnbeløpsdato: LocalDate? = null,
```

Comment:
Trenger vi egentlig to opplysninger her:
- Dato for g-omregning
- Versjon som ble brukt i g-omregning? 

Bare dato er vel litt tynt når ting skjer i skjøringspunktet prodsatt, men ikke g-omregnet?

Reply:
Dette er tenkt å være datoen for g-versjon, og er satt default i databasen til 2021-05-01, siden alle vi har i dag er forrige år sin versjon. Tanken er at vi setter versjon av g-beløp som er brukt i behandlingen og oppdaterer ved omregning.

Reply:
Tanken er å sette alle til nyeste G-omregning i db. Med g-omregning feltet er not null og default 01-05-2021.
Dersom det opprettes nye behandlinger blir også de satt til g-omregnet 01-05-2021.
Kan si meg enig i at TilkjentYtelse kan ha informasjonen om hvilken g-omregning som gjelder, altså funksjonell dato 1. mai.
Så kan sistEndret vise når den ble endret. 
Vi har planer om å lage en task pr behandling/tilkjentytelse - tenker at taskrammeverket kan ha ansvaret for rekjøring ved feil.
Dersom det ikke skjer en feil og den kjører i gjennom er det jo egentlig ikke ønskelig å kjøre igjen. Ved ny G-omregning vil altså datoen bli 01.05.2022 og det skal dermed ikke være mulig å kjøre på nytt.
Siden det er en dato kan det beregnes g-beløp senere i året.


""Å bruke dato for fra hvilket dato g-beløpet er gjeldende, sier ikke noe om hvilken versjon av denne som er i bruk, hvis man då bruker en string så kan man rekjøre den samme dagen, og også ha litt mer info om hvilken versjon som er brukt."
Hva mener du med dette? Dersom gjeldende dato er satt til 01-05-2021 så er det G-omregnet korrekt frem til denne datoen. Igjen så har vi task-rammeverket dersom det skulle bli en feil som kaster exception.
Eller tenker du på om det kjøres OK, men at selve beregningen har en bug som resulterer i feil beløp?

Reply:
> Dette er tenkt å være datoen for g-versjon, og er satt default i databasen til 2021-05-01, siden alle vi har i dag er forrige år sin versjon. Tanken er at vi setter versjon av g-beløp som er brukt i behandlingen og oppdaterer ved omregning.

👍  Burde ikke default settes på behandlingen då også, ettersom den er `NOT NULL`? 

Hvis datoet er datoet for når g-beløpet er gjeldende, og man trenger å kjøre g-beregning på nytt så er det fortsatt det samme datoet, sånn att sette datoet sier ikke noe om hvilken "version" som ble brukt, kun fra hvilket dato g-beløpet er gjelende. 
Hvordan gjør man då hvis man må gjøre en ny g-beregning? Eks pga noe som er feil

Reply:
Jo, sant, burde sette default dato i behandling også, slik at den er non-null der også. Rent teknisk funker det vel nå, men blir bedre å unngå null-verdi i koden også.

Jeg ser den. Tenkte kanskje det holdt å bruke informasjonen som ligger i Sporbar-objektet i de tilfellene. Der har vi endret av og endret tidspunkt. Men jeg innser at da måtte vi hatt egen "saksbehandler"-navn for G-omregning, og at det er ryddigst med en kolonne til :)

Reply:
Når den er `NOT NULL` i databasen og `null` i domeneobjektet så tryner det når man tar insert, default verdi i databasen brukes bare hvis den ikke settes in en `insert into foo (value1, value2)` :)